bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/sitequality/SiteQualityPanel.java

```diff
@@ -166,43 +164,59 @@
 				weighty = 50;
 			}});
 
+            final Frame sqpFrame = (Frame) SwingUtilities.getWindowAncestor(this);
+
             add(new Box(BoxLayout.LINE_AXIS) {{
 				add(new JButton(Resources.getIcon("eclipse/add.gif")) {{
                     setToolTipText("Add new timing window");
                     setFocusable(false);
-                    addActionListener(e -> {
-						TimingWindow tw = new TimingWindow();
-						TimingWindowDialog twd = new TimingWindowDialog((Frame) SwingUtilities.getWindowAncestor(SiteQualityPanel.this));
-						tw = twd.showEdit(tw);
-						if (tw != null) {
+                    addActionListener(e ->
+						new TimingWindowDialog(sqpFrame).openNew().foreach(tw -> {
 							owner.getDataObject().addTimingWindow(tw);
 							int index = table.getModel().getRowCount() - 1;
-							table.changeSelection(index, 0, false, false); // magic
-						}
-					});
+							table.changeSelection(index, 0, false, false);
+						}));
                     ButtonFlattener.flatten(this);
 				}});
 
+                add(new JButton(Resources.getIcon("eclipse/download.gif")) {{
+                    setToolTipText("Import timing windows from text file");
+                    setFocusable(false);
+                    addActionListener(e -> {
+                        final TimingWindowImporter importer = new TimingWindowImporter(SiteQualityPanel.this);
+						final TimingWindowParser.TimingWindowParseResults results = importer.openImport();
```

Comment:
It will block until the dialog is closed, and the dialog is modal, so the OT will require you to dismiss the dialog before you can do anything else in the OT.

This is pretty standard for any `Dialog` or `JDialog`, and blocks when you call `setVisible(true)`, which is all that I do in `openImport` to show the dialog.